File: springboot-grpc/domain/src/main/java/com/scriptonbasestar/domain/model/UserEntity.java
package com.scriptonbasestar.domain.model;

import com.scriptonbasestar.domain.type.UserGradeType;
import lombok.*;

import javax.persistence.*;

/**
 * @author chaeeung.e
 * @since 2017-06-27
 */
@Entity(name = "UserEntity")
@Table(name = "T_USER")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
@EqualsAndHashCode
@ToString
public class UserEntity {

	@Id
	@GeneratedValue
	private Long id;

	@Column(length = 100)
	private String name;

	@Column(length = 20)
	@Enumerated(EnumType.STRING)
	private UserGradeType userGradeType;

}

File: 2022-webmvc/domain/auth/src/main/kotlin/org/scriptonbasestar/domain/auth/persistence/UserEntity.kt
package org.scriptonbasestar.domain.auth.persistence

import org.hibernate.annotations.Nationalized
import org.scriptonbasestar.base.BaseSeqUuidEntity
import javax.persistence.Entity

@Entity
class UserEntity(
    @Nationalized val name: String,
    @Nationalized val description: String,
    val enabled: Boolean,
) : BaseSeqUuidEntity()
